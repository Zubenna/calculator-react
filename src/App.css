.App {
  text-align: center;
}

.row {
  display: flex;
}

.calc-container {
  justify-content: center;
  align-content: center;
  width: 15rem;
  height: 20rem;
  margin: 0 auto;
}

.display {
  width: 100%;
  text-align: right;
  box-sizing: border-box;
  height: 2.5rem;
  font-size: 1.5rem;
  border-top-left-radius: 5px;
  border-top-right-radius: 5px;
}
