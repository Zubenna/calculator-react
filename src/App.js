import './App.css';

function App() {
  return (
    <div className="App">
      <h1>Testing React Calculator Project Setup...Testing Deployment</h1>
    </div>
  );
}

export default App;
