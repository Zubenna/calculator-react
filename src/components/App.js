import React, { useState } from 'react';
import '../App.css';
import ButtonPanel from './ButtonPanel';
import Display from './Display';
import calculate from '../logic/calculate';

const App = () => {
  const [obj, setObj] = useState(
    {
      total: null,
      operation: null,
      next: null,
    },
  );

  const handleClick = buttonName => {
    const changedState = calculate(obj, buttonName);
    setObj(changedState);
  };

  return (
    <div className="App">
      <h1 className="app-head">Simple Calculator; Final Version</h1>
      <section className="calc-container">
        <>
          <Display result={obj.next || obj.operation || obj.total || 0} />
          <ButtonPanel clickHandler={handleClick} />
        </>
      </section>
    </div>
  );
};
export default App;
