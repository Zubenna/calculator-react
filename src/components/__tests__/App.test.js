import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';

describe('App', () => {
  it('Tests the rendering of components', () => {
    const tree = renderer
      .create(<App />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
