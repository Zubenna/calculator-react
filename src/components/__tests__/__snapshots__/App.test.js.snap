// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Tests the rendering of components 1`] = `
<div
  className="App"
>
  <h1
    className="app-head"
  >
    Simple Calculator; Final Version
  </h1>
  <section
    className="calc-container"
  >
    <div
      className="display"
    >
      <p>
        0
      </p>
    </div>
    <div
      className="row"
    >
      <button
        className="btn "
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn "
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn "
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn colorBtn"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="btn "
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn "
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn "
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn colorBtn"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="btn "
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn "
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn "
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn colorBtn"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="btn "
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn "
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn "
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn colorBtn"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="zeroBtn "
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn "
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn colorBtn"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </section>
</div>
`;
